File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXSlider.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.geometry.Orientation;
import javafx.scene.control.Slider;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Name(JavaFXExtension.NS + "UXSlider")
public class UXSlider extends UXControl<Slider> {
    public UXSlider(Environment env, Slider wrappedObject) {
        super(env, wrappedObject);
    }

    public UXSlider(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    interface WrappedInterface {
        @Property Orientation orientation();
        @Property boolean showTickMarks();
        @Property boolean showTickLabels();
        @Property boolean snapToTicks();

        @Property double blockIncrement();
        @Property int minorTickCount();
        @Property double majorTickUnit();

        @Property double min();
        @Property double max();
        @Property double value();
    }

    @Signature
    public void __construct() {
        __wrappedObject = new Slider();
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/UXSlider.php
<?php
namespace php\gui;

/**
 * Class UXSlider
 * @package php\gui
 * @packages gui, javafx
 */
class UXSlider extends UXControl
{
    /**
     * @var double
     */
    public $min;

    /**
     * @var double
     */
    public $max;

    /**
     * @var double
     */
    public $value;

    /**
     * @var double
     */
    public $blockIncrement;

    /**
     * @var bool
     */
    public $showTickMarks;

    /**
     * @var bool
     */
    public $showTickLabels;

    /**
     * @var bool
     */
    public $snapToTicks;

    /**
     * @var double
     */
    public $majorTickUnit = 25;

    /**
     * @var int
     */
    public $minorTickUnit = 0;

    /**
     * @var string
     */
    public $orientation = 'HORIZONTAL';
}